File: app/models/req_info.rb
class ReqInfo < ApplicationRecord
  belongs_to :request
  after_create :req_info_send

  def req_info_send
    ActionCable.server.broadcast "req_info_channel",
                                 req_id: self.request.id,
                                 path_url: self.request.path_url,
                                 req_inf_id: self.id,
                                 remote_ip: self.remote_ip,
                                 req_method: self.req_method,
                                 scheme: self.scheme,
                                 query_string: self.query_string,
                                 query_params: self.query_params,
                                 cookies: self.cookies,
                                 headers: self.headers
    # head :ok
  end
end

File: app/assets/javascripts/channels/req_info.js
//= require cable
//= require_self
//= require_tree .

this.App = {};

App.cable = ActionCable.createConsumer();

App.trap = App.cable.subscriptions.create('ReqInfoChannel', {
    received: function (data) {
        var req_info_id = $('#req_info').attr('data-id');
        // console.log(data);
        if (req_info_id == 'all'){
            addReqInfo(data);
        }else {
            if (req_info_id == data.req_id){
                addReqInfo(data);
            }
        }
    }
});

function addReqInfo(data) {
    $('#req_info tbody').append(
        '<tr><td>'+ data.path_url +'</td><td>'+ data.remote_ip + '</td><td>'+ data.req_method +'</td><td>'+ data.scheme +'</td><td>'+ data.query_string +'</td><td>'+ data.query_params +'</td><td>'+ data.cookies + '</td><td>'+ data.headers +'</td></tr>'
    );
}
